.github/workflows/build-test.yml

```diff
@@ -0,0 +1,78 @@
+name: Build and Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    types: [opened, synchronize]
+
+jobs:
+  build_image:
+    name: Build image
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        architecture: [amd64, arm64]
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build the image
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          platforms: linux/${{ matrix.architecture }}
+          push: false
+          tags: ghcr.io/${{ github.repository_owner }}/bettergov:latest
+
+  lint_checks:
+    name: Lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 24
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run ESLint
+        run: npm run lint
+
+  e2e_tests:
+    name: Playwright Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 24
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Install Playwright browsers
+        run: npx playwright install --with-deps
+
+      - name: Run E2E Tests
+        run: npm run test:e2e
```

Comment:
This is not that urgent and can be addressed in a separate PR, but these end-to-end tests with Playwright are taking a pretty long time to run. So far, the E2E tests step is taking at least 17 minutes. Do you think there's a way to speed up running these E2E tests?

I found a documentation page on Playwright about sharding on GitHub Actions that might really help us: [Sharding | Playwright](https://playwright.dev/docs/test-sharding#github-actions-example)